app/include/zmk/endpoints.h

```diff
@@ -6,10 +6,66 @@
 
 #pragma once
 
+#include <zmk/ble.h>
 #include <zmk/endpoints_types.h>
 
-int zmk_endpoints_select(enum zmk_endpoint endpoint);
-int zmk_endpoints_toggle();
-enum zmk_endpoint zmk_endpoints_selected();
+/**
+ * Recommended length of string buffer for printing endpoint identifiers.
+ */
+#define ZMK_ENDPOINT_STR_LEN 10
+
+#ifdef CONFIG_ZMK_USB
+#define ZMK_ENDPOINT_USB_COUNT 1
+#else
+#define ZMK_ENDPOINT_USB_COUNT 0
+#endif
+
+#ifdef CONFIG_ZMK_BLE
+#define ZMK_ENDPOINT_BLE_COUNT ZMK_BLE_PROFILE_COUNT
+#else
+#define ZMK_ENDPOINT_BLE_COUNT 0
+#endif
+
+/**
+ * The total number of different (struct zmk_endpoint_instance) values that can
+ * be selected.
+ *
+ * Note that this value may change between firmware versions, so it should not
+ * be used in any persistent storage.
+ */
+#define ZMK_ENDPOINT_COUNT (ZMK_ENDPOINT_USB_COUNT + ZMK_ENDPOINT_BLE_COUNT)
+
+bool zmk_endpoint_instance_equals(struct zmk_endpoint_instance a, struct zmk_endpoint_instance b);
```

Comment:
So, Zephyr uses `eq` shorthand pretty consistently, any reason not to use that convention as well here?

Reply:
Nope. Wasn't aware of that convention. I can update this.

Reply:
Done. Also changed the `_print()` function to `_to_str()` and reordered the arguments to match what looks like Zephyr conventions for that type of function as well.